### PR TITLE
Add missing header guards in android api.h

### LINKED ISSUE #36324: Missing header guard in platform/android/api/api.h
**Godot version:**
4.0 master

**Issue description:**
Missing header guard in android/api/api.h. Detected via [LGTM](https://lgtm.com/projects/g/godotengine/godot/snapshot/be834fecef2e4d398cb9570d23e8f3b9f5d61f39/files/platform/android/api/api.h?sort=name&dir=ASC&mode=heatmap#x239b4089bba58f53:1) on the main README file. 
File: https://github.com/godotengine/godot/blob/master/platform/android/api/api.h

### PR DESCRIPTION
**Pull request by twaritwaikar**
Fixes #36324 
Minor addition doesn't affect the build at all.

### CODE REVIEW

File: platform/android/api/api.h [hunk 1]
```diff
@@ -28,5 +28,10 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
+#ifndef API_H
```
Comment (akien-mga) [inline platform/android/api/api.h]: I would name it ANDROID_API_H to ensure uniqueness. There's also a similar one for JavaScript that might be worth checking.

Comment (akien-mga) [reply platform/android/api/api.h]: I checked and `jplatform/javascript/api/api.h` has the same problem indeed.

**Comment by IronicallySerious** [comment]
These are strictly following the filename convention for header guards and
I think most IDE do that too. This might be a problem so I can do a full
codebase scan on colliding header guards and fix them.

**Comment by akien-mga** [comment]
Well it's the same with `platform/*/platform_config.h` and `platform/*/export/export.h`, which currently don't have header guards. If we add them, they should be fully specified to disambiguate.

**Comment by akien-mga** [comment]
Here's a diff between a list of all non-thirdparty header files, and the list listed with duplicates removed. So the removed lines in the diff identify files in our codebase that have the same filename, and whose guards should likely be reviewed/disambiguated. Most are either without guards as mentioned in the previous comment, or in `gdnative/include` where they seem to use the `GODOT_` prefix.

**Comment by lawnjelly** [comment]
This is actually a good reason for the use of `#pragma once`, a microsoft idea that surprisingly(!) actually makes good sense. It is actually simpler and better imo than named guards, unless the guards are being used for specific custom multi-include purposes (e.g. header does something different depending on defines).

And I see Calinou has a PR for this purpose: #34143

**Comment by akien-mga** [comment]
Superseded by #37219.

**Pull request closed without merging**